test/linalg/symmetric.jl

```diff
@@ -42,206 +41,229 @@ end
-        end
-
-        # transpose, ctranspose
-        S = Symmetric(asym)
-        H = Hermitian(asym)
-        if eltya <: Real
-            @test  transpose(S) === S == asym
-            @test ctranspose(S) === S == asym
-            @test  transpose(H) === H == asym
-            @test ctranspose(H) === H == asym
-        else
-            @test  transpose(S) === S
-            @test ctranspose(S) ==  Symmetric(conj(asym))
-            @test  transpose(H) ==  Hermitian(transpose(asym))
-            @test ctranspose(H) === H == asym
-        end
-
-        #tril/triu
-        for di in -n:n
-            @test triu(Symmetric(a+a.'),di) == triu(a+a.',di)
-            @test tril(Symmetric(a+a.'),di) == tril(a+a.',di)
-            @test triu(Hermitian(asym),di) == triu(asym,di)
-            @test tril(Hermitian(asym),di) == tril(asym,di)
-            @test triu(Symmetric(a+a.',:L),di) == triu(a+a.',di)
-            @test tril(Symmetric(a+a.',:L),di) == tril(a+a.',di)
-            @test triu(Hermitian(asym,:L),di) == triu(asym,di)
-            @test tril(Hermitian(asym,:L),di) == tril(asym,di)
-        end
-
-        eltya == BigFloat && continue # Revisit when implemented in julia
-        d, v = eig(asym)
-        @test asym*v[:,1] ≈ d[1]*v[:,1]
-        @test v*Diagonal(d)*v' ≈ asym
-        @test isequal(eigvals(asym[1]), eigvals(asym[1:1,1:1]))
-        @test abs.(eigfact(Hermitian(asym), 1:2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
-        eig(Hermitian(asym), 1:2) # same result, but checks that method works
-        @test abs.(eigfact(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2)[:vectors]'v[:,1:2]) ≈ eye(eltya, 2)
-        eig(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) # same result, but checks that method works
-        @test eigvals(Hermitian(asym), 1:2) ≈ d[1:2]
-        @test eigvals(Hermitian(asym), d[1] - 1, (d[2] + d[3])/2) ≈ d[1:2]
-        @test full(eigfact(asym)) ≈ asym
-        @test eigvecs(Hermitian(asym)) ≈ eigvecs(asym)
-
-        # relation to svdvals
-        @test sum(sort(abs.(eigvals(Hermitian(asym))))) == sum(sort(svdvals(Hermitian(asym))))
-
-        # cond
-        @test cond(Hermitian(asym)) ≈ cond(asym)
-
-        # det
-        @test det(asym) ≈ det(Hermitian(asym, :U))
-        @test det(asym) ≈ det(Hermitian(asym, :L))
-        if eltya <: Real
-            @test det(asym) ≈ det(Symmetric(asym, :U))
-            @test det(asym) ≈ det(Symmetric(asym, :L))
-        end
-        @test det(a + a.') ≈ det(Symmetric(a + a.', :U))
-        @test det(a + a.') ≈ det(Symmetric(a + a.', :L))
-
-        # isposdef[!]
-        @test isposdef(Symmetric(asym)) == isposdef(full(Symmetric(asym)))
-        @test isposdef(Hermitian(asym)) == isposdef(full(Hermitian(asym)))
-        if eltya != Int
-            @test isposdef!(Symmetric(copy(asym))) == isposdef(full(Symmetric(asym)))
-            @test isposdef!(Hermitian(copy(asym))) == isposdef(full(Hermitian(asym)))
-        end
+            @testset "conversion" begin
```

Comment:
I think some of the `if`s in this testset can be simplified with the new `asym` definition too?